// get-md-date/index.es5.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _commonmarkHelpers = require('commonmark-helpers');

var _trimHtmlTag = require('trim-html-tag');

var _trimHtmlTag2 = _interopRequireDefault(_trimHtmlTag);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var isDate = function isDate(format, locale, node) {
  return (0, _moment2['default'])((0, _commonmarkHelpers.text)(node), format, locale, true).isValid();
};

exports['default'] = function (format, locale, input) {
  var node = (0, _commonmarkHelpers.match)(input, function (node) {
    return isDate(format, locale, node);
  });
  if (!node) return;
  return {
    text: (0, _commonmarkHelpers.text)(node),
    html: (0, _trimHtmlTag2['default'])((0, _commonmarkHelpers.html)(node)),
    unix: _moment2['default'].utc((0, _commonmarkHelpers.text)(node), format, locale, true).unix(),
    moment: (0, _moment2['default'])((0, _commonmarkHelpers.text)(node), format, locale, true),
    node: node
  };
};

module.exports = exports['default'];
